boards/PDM/Src/Io/Io_Efuse.c

```diff
@@ -0,0 +1,350 @@
+    // Place the Status Register address into bits 10->13
+    serial_input_data =
+        (uint16_t)((SI_STATR_0_ADDR & EFUSE_ADDR_MASK) << EFUSE_ADDR_SHIFT);
+
+    // Shift bit 3 of the address (SOA3: the channel number) to the 13th bit
+    serial_input_data = (uint16_t)(
+        serial_input_data |
+        ((register_address & SOA3_MASK) << EFUSE_ADDR_SHIFT) |
+        (register_address & (SOA2_MASK | SOA1_MASK | SOA0_MASK)));
+
+    // Invert watchdog bit state (Note: It is safe to do so even if the watchdog
+    // is disabled)
+    if (e_fuse->wdin_bit_to_set)
+    {
+        SET_BIT(serial_input_data, WATCHDOG_BIT);
+    }
+    else
+    {
+        CLEAR_BIT(serial_input_data, WATCHDOG_BIT);
+    }
+
+    // Invert watchdog bit state for next write
+    e_fuse->wdin_bit_to_set = !e_fuse->wdin_bit_to_set;
+
+    Io_Efuse_CalculateParityBit(&serial_input_data);
+
+    ExitCode exit_code = Io_Efuse_ReadFromEfuse(
+        &serial_input_data, register_value, e_fuse->efuse_spi_handle,
+        e_fuse->chip_select_port, e_fuse->chip_select_pin);
+
+    // Only return register contents and clear bits 9->15
+    *register_value = READ_BIT(*register_value, EFUSE_SO_DATA_MASK);
+
+    return exit_code;
+}
+
+static void Io_Efuse_CalculateParityBit(uint16_t *serial_data_input)
+{
+    uint16_t _serial_data_input = *serial_data_input;
+    bool     parity_bit;
+
+    for (parity_bit = 0; _serial_data_input > 0; _serial_data_input >>= 1)
+    {
+        parity_bit ^= READ_BIT(_serial_data_input, 1U);
+    }
+
+    if (parity_bit)
+    {
+        SET_BIT(*serial_data_input, PARITY_BIT);
+    }
+    else
+    {
+        CLEAR_BIT(*serial_data_input, PARITY_BIT);
+    }
+}
+
+static ExitCode Io_Efuse_WriteToEfuse(
+    uint16_t *         tx_data,
+    SPI_HandleTypeDef *efuse_spi_handle,
+    GPIO_TypeDef *     chip_select_port,
+    uint16_t           chip_select_pin)
+{
+    HAL_StatusTypeDef status = HAL_OK;
+
+    HAL_GPIO_WritePin(chip_select_port, chip_select_pin, GPIO_PIN_RESET);
+    status = HAL_SPI_TransmitReceive(
+        efuse_spi_handle, (uint8_t *)tx_data, (uint8_t *)tx_data, 1U, 100U);
+    HAL_GPIO_WritePin(chip_select_port, chip_select_pin, GPIO_PIN_SET);
+
+    if (status != HAL_OK)
+    {
+        return EXIT_CODE_TIMEOUT;
+    }
+
+    return EXIT_CODE_OK;
+}
+
+static ExitCode Io_Efuse_ReadFromEfuse(
+    uint16_t *         tx_data,
+    uint16_t *         rx_data,
+    SPI_HandleTypeDef *efuse_spi_handle,
+    GPIO_TypeDef *     chip_select_port,
+    uint16_t           chip_select_pin)
+{
+    HAL_StatusTypeDef status = HAL_OK;
+
+    // Send the command stored in tx_data to the status register, to read the
+    // data from the register address specified in tx_data
+    HAL_GPIO_WritePin(chip_select_port, chip_select_pin, GPIO_PIN_RESET);
+    status = HAL_SPI_TransmitReceive(
+        efuse_spi_handle, (uint8_t *)tx_data, (uint8_t *)rx_data, 1U, 100U);
+    HAL_GPIO_WritePin(chip_select_port, chip_select_pin, GPIO_PIN_SET);
```

Comment:
Just out of curiosity did you happen to test this function without toggling the nss pin high here? Just wondering b/c when I did a write read with the LTC chip SS was held low until the read was complete.

![image](https://user-images.githubusercontent.com/32230980/93818955-e979bd00-fc0f-11ea-989b-cb98cde4c6d1.png)

Reply:
Yes I did and it didn't work, I believe there's a comment in Io_efuse.h that says it must be toggled